microproxy/layer/application/tls.py

```diff
@@ -181,3 +186,10 @@ def process_and_return_context(self):
             self.context.scheme = "https"
 
         raise gen.Return(self.context)
+
+    @gen.coroutine
+    def process_replay(self):
+        dest_ssl_ctx = self.create_dest_sslcontext()
```

Comment:
Maybe we can move out some of the tls layer code to utils function. So that if you want to create a new layer, we can utilize it.